public/modules/moduleMetadataService.js

```diff
@@ -1,8 +1,8 @@
 angular.module('newApp')
 .factory('moduleMetadata', function($http) {
     return {
-        get : function() {
-            return $http.get('/api/units');
+        get : function(moduleID) {
+            return $http.get('/api/units' + (moduleID ? '/' + moduleID : ''));
```

Comment:
Ok, I've created getAll() as an additional/separate method.  

Does that fix the last issue about promise vs array?  If not, I don't think I understand the issue you're describing.